src/app/region.js

```diff
@@ -0,0 +1,188 @@
+  })
+  // Coordinates have array of lines
+  // Lines have array of points
+  // Points have one array of latlngs [lng, lat]
+  for (let lineIndex = coordinates.length - 1; lineIndex >= 0; lineIndex--) {
+    const line = coordinates[lineIndex]
+    for (let pointsIndex = line.length - 1; pointsIndex >= 0; pointsIndex--) {
+      const points = line[pointsIndex]
+      for (let coordIndex = points.length - 1; coordIndex >= 0; coordIndex--) {
+        const point = points[coordIndex]
+        const lat = point[1]
+        const lng = point[0]
+        // Checking if latlng is within bounding box
+        // If not remove from points
+        if (lng < Number(bounds.west) - buffer || lng > Number(bounds.east) + buffer || lat < Number(bounds.south) - buffer || lat > Number(bounds.north) + buffer) {
+          points.splice(coordIndex, 1)
+        }
+      }
+      // If no points in line, remove from line
+      if (points.length === 0) { line.splice(pointsIndex, 1) }
+    }
+    // If no lines in coordinates, remove from coordinates
+    if (line.length === 0) { coordinates.splice(lineIndex, 1) }
+  }
+
+  // If no coordinates, remove entire feature from array of features
+  for (let i = features.length - 1; i >= 0; i--) {
+    const feature = features[i]
+    if (feature.geometry.coordinates.length === 0) { features.splice(i, 1) }
+  }
+}
+
+function getBboxArea (bounds) {
+  const width = (bounds.east - bounds.west)
+  const height = (bounds.north - bounds.south)
+  const area = width * height
+  return area
+}
+
+export function showRegion (bounds) {
+  // If bounds are cleared, remove data source from tangram
+  if (!bounds) {
+    const scene = getCurrentScene()
+    delete scene.sources.routes
+    setCurrentScene(scene)
+    return
+  }
+
+  // If area of bounding box exceeds max_area, display error
+  // Arbitrary number for max_area, edit if needed
+  const maxArea = 0.01
+  const area = getBboxArea(bounds)
+  if (area > maxArea) {
+    const message = 'Please zoom in and reduce the size of your bounding box'
+    store.dispatch(setRouteError(message))
+    return
+  }
+
+  // First, convert bounds to waypoints
+  // This way we can get the bounding box from valhalla and suffixes
+  const waypoints = [L.latLng(bounds.south, bounds.west), L.latLng(bounds.north, bounds.east)]
+  // Go get route using Valhalla
+  getRoute(host, waypoints)
+    // After getting route, get the bounding box and the relevant tiles from box
+    .then((response) => {
+      const bbox = response.trip.summary
+      const suffixes = getSuffixes(bbox)
+      return suffixes
+    })
+    // Second, fetch the OSMLR Geometry tiles using the suffixes
+    .then((suffixes) => {
+      store.dispatch(startLoading())
+      fetchOSMLRGeometryTiles(suffixes)
+        .then((results) => {
+          const features = results.features
+          // Remove from geojson, routes outside bounding box (bounds)
+          withinBbox(features, bounds)
```

Comment:
let's rewrite this so that it doesn't depend on `withinBbox` doing work on `features` as a side effect. Even if it really does (because of how objects are passed around by reference) we should prepare for a future in which it might not. The `withinBbox()` function should be revised so that it returns a result, and here, the result should be assigned to a variable.